botocore/session.py

```diff
@@ -858,6 +863,15 @@ def _resolve_region_name(self, region_name, config):
         self._last_client_region_used = region_name
         return region_name
 
+    def _validate_region_name(self, region_name):
+        """Provided region_name must be a valid host label."""
+        if region_name is None:
+            return
+        valid_host_label = re.compile(r'^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{,63}(?<!-)$')
```

Comment:
Do you think it makes sense to hoist this up to the class? Or maybe even create a utility function for doing host label validation?

Reply:
Yeah, I was originally thinking about having it live in utils but didn't know if we wanted it public. I'll move it over there really quick.